Rakefile

```diff
@@ -123,11 +123,12 @@ maybe_credentials = "#{ENV['GIT_USER']}:#{ENV['GIT_PASSWORD']}@" if ENV['GIT_USE
     add_files: tini_and_gosu_add_file_meta,
     create_user_and_group: create_user_and_group_cmd,
     before_install: [
-      "echo 'deb http://ppa.launchpad.net/webupd8team/java/ubuntu xenial main' > /etc/apt/sources.list.d/webupd8team-java.list",
-      'apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys EEA14886',
       'apt-get update',
-      'echo oracle-java8-installer shared/accepted-oracle-license-v1-1 select true | /usr/bin/debconf-set-selections',
-      'apt-get install -y oracle-java8-installer git subversion mercurial openssh-client bash unzip curl',
+      'apt-get install -y git subversion mercurial openssh-client bash unzip curl',
+      'mkdir jre',
+      'curl -L -v --cookie "oraclelicense=accept-securebackup-cookie" http://download.oracle.com/otn-pub/java/jdk/8u171-b11/512cd62ec5174c3487ac17c61aaa89e8/jre-8u171-linux-x64.tar.gz > jre-8u171-linux-x64.tar.gz && gunzip jre-8u171-linux-x64.tar.gz && tar xvf jre-8u171-linux-x64.tar -C /jre --strip-components=1',
```

Comment:
You should be able to pipe the `curl` to `tar` without an intermediate file in between.